File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) return false;
        unordered_map<char,int> mp;
        for(char& ch : s)
            mp[ch]++;
        for(char& ch : t)
        {
            if(mp[ch] == 0) return false;
            if(mp.find(ch) == mp.end())
                return false;
            mp[ch]--;
        }
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        dt = {}
        for char in s:
            if dt.get(char):
                dt[char] += 1
            else:
                dt[char] = 1
        for char in t:
            if char not in dt:
                return False
            dt[char] -= 1
            if dt.get(char) == 0:
                dt.pop(char)
        return True